### PR TITLE
Add in ssh port option

### PR DESCRIPTION
**Pull request by nealfennimore**
Allows for rebuilding a remote host on a non-standard ssh port

### CODE REVIEW

File: src/nixos-anywhere.sh [hunk 1]
```diff
@@ -74,6 +77,10 @@ while [[ $# -gt 0 ]]; do
     ssh_private_key_file=$2
     shift
     ;;
+  -p)
+    ssh_port=$2
```
Comment (Kranzes) [inline src/nixos-anywhere.sh:81]: It's best to not hardcode 22. Some people have it set to something else in their ssh config.

Comment (phaer) [reply src/nixos-anywhere.sh:81]: Yes, the option should only be passed if explicitly set.

And while we are at it, maybe we could just add `$SSH_ARGS`or so to allow passing `-p` and various `-o` options or would this complicate things here?

Comment (nealfennimore) [reply src/nixos-anywhere.sh]: @phaer @Kranzes  I added in the option for passing in ssh port and option via the `-p` and `-o` arg. Let me know if that works

**Review by phaer — APPROVED** [review]
Thanks, LGTM!

**Comment by Kranzes** [comment]
i think `-o` and `-p` are too generic for being only for ssh.

**Comment by nealfennimore** [comment]
> i think `-o` and `-p` are too generic for being only for ssh.

How about `--ssh-option` and `--ssh-port`?

**Comment by Lassulus** [comment]
we need to set this port also for the `--build-on-remote` option, I propose adding the port to a separate variable so we can use it in the `--builders` arg if it's actually set. this would also make the filtering out part easier.

Also I'm fine with using -p and -o since ssh is the thing we mostly want to configure. adding the long options additionally would be nice though.

File: src/nixos-anywhere.sh [hunk 2]
```diff
@@ -314,13 +328,21 @@ SSH
 TMPDIR=/root/kexec setsid ${maybe_sudo} /root/kexec/kexec/run
 SSH
 
-  # wait for machine to become unreachable
+  # use the default SSH port to connect at this point
```
[the remark this answers is not in the document]
Comment (Mic92) [reply src/nixos-anywhere.sh:331]: There is also port forwarding i.e. if the machine is behind a router or if you use qemu where the port used on the system is not the same as the port that you need to connect to.

**Comment by Mic92** [comment]
Just saw this PR. I have in the meantime merged this: https://github.com/numtide/nixos-anywhere/pull/130/files
I think we need a combination of both because ssh ports can be both specified in configuration and manually.

File: src/nixos-anywhere.sh [hunk 3]
```diff
@@ -11,6 +11,10 @@ Options:
   set the flake to install the system from.
 * -i <identity_file>
   selects which SSH private key file to use.
+* -p, --ssh-port <ssh_port>
+  set the ssh port to connect with
+* -o, --ssh-option <ssh_option>
```
Comment (Mic92) [inline src/nixos-anywhere.sh:16]: I think I want to reserve -o for something else, it's not super descriptive on its own.

```suggestion
* --ssh-option <ssh_option>
```

File: src/nixos-anywhere.sh [hunk 4]
```diff
@@ -74,6 +82,14 @@ while [[ $# -gt 0 ]]; do
     ssh_private_key_file=$2
     shift
     ;;
+  -p | --ssh-port)
+    ssh_args+=("-p" "$2")
+    shift
+    ;;
+  -o | --ssh-option)
```
Comment (Mic92) [inline src/nixos-anywhere.sh:89]:
```suggestion
  --ssh-option)
```

**Comment by Lassulus** [comment]
obsolete by https://github.com/numtide/nixos-anywhere/pull/153

**Pull request closed without merging**